lightning/src/routing/scoring.rs

```diff
@@ -460,25 +466,420 @@ impl<T: Time> Readable for ChannelFailure<T> {
 	}
 }
 
+/// [`Score`] implementation using channel success probability distributions.
+///
+/// Based on *Optimally Reliable & Cheap Payment Flows on the Lightning Network* by Rene Pickhardt
+/// and Stefan Richter [[1]]. Given the uncertainty of channel liquidity balances, probability
+/// distributions are defined based on knowledge learned from successful and unsuccessful attempts.
+/// Then the negative log of the success probability is used to determine the cost of routing a
+/// specific HTLC amount through a channel.
+///
+/// Knowledge about channel liquidity balances takes the form of upper and lower bounds on the
+/// possible liquidity. Certainty of the bounds is decreased over time using a decay function. See
+/// [`ProbabilisticScoringParameters`] for details.
+///
```

Comment:
Thanks for the update. I think its still a little vague, I suppose the decaying is also bad for long-term learning, but my other long-term-learning concern here is just about how we go about learning to begin with. Our whole thing here is about trying to remember the *current* channel balances on every channel on the network. We don't so much learn anything as we do try to keep a mental map of the whole network up-to-date all the time. That's great if you're probing or have payment flows, but otherwise we're really not learning at all, IMO.

Reply:
Fully agree! Those seem to be two orthogonal problems.

1. The probability that liquidity is available (as a probabilistic model for the reliability) 
2. The actual observed reliability of a specific node / channel (as a provenance score or distribution) 

I claim - but obviously don't know - that the first one will be much more impactful. In particular for the first one we know how to properly quantify this as the probabilistic model gives us the propper information theoretic framework.

That being said I am very interested in examining the second one and investigating if it is useful to include it to the cost function. I have drafts for an approach for assigning provenance scores as a probability distribution that can be used to create the joint distribution between this and the liquidity based model. The good thing is that models for the second question can be learnt from log files and also tested against the log file. Will be very happy to collaborate on this in particular as you seem to have access to data / logfiles

Reply:
Right, I assume that both will be important (though somewhat annoyingly the lightning onion errors dont tell us which happened when we get a fail - liquidity limit or node offline), but either way we don't yet do either and that would be bad for some users :)

Reply:
Re-worded the first part of the sentence as discussed offline.

Reply:
Having two opposing conjectures / hypothesis is one of the best starting points for research. As said I am very interested in these two question (1. How to define a reliability based probabilistic provenance score for nodes / channels 2. How useful will this be in the cost function) and happy to collaborate and find this out. Also very happy to at least anecdotaly be proven wrong on my conjecture